docs/content/configuration/index.md

```diff
@@ -222,7 +222,7 @@ The Druid servers [emit various metrics](../operations/metrics.html) and alerts
 |`druid.emitter.http.basicAuthentication`|Login and password for authentification in "login:password" form, e. g. `druid.emitter.http.basicAuthentication=admin:adminpassword`|not specified = no authentification|
 |`druid.emitter.http.flushTimeOut`|The timeout after which an event should be sent to the endpoint, even if internal buffers are not filled, in milliseconds.|not specified = no timeout|
 |`druid.emitter.http.batchingStrategy`|The strategy of how the batch is formatted. "ARRAY" means `[event1,event2]`, "NEWLINES" means `event1\nevent2`, ONLY_EVENTS means `event1event2`.|ARRAY|
-|`druid.emitter.http.maxBatchSize`|The maximum batch size, in bytes.|5191680 (i. e. 5 MB)|
+|`druid.emitter.http.maxBatchSize`|The maximum batch size, in bytes.|the minimum of (10% of JVM heap size divided by 50) or (5191680 (i. e. 5 MB))|
```

Comment:
In memory constrained settings I think it would be better to limit the batchQueueSizeLimit rather than the maxBatchSize. It's because very small batches might be an inefficient way to transfer metrics to the http server. We might want to limit both though. It's common to set Xmx to something small (like 64MB) for middleManagers, and in that case, probably 2 batches in the queue * 2.5 MB per batch makes sense.

Reply:
adjusted this to lower the queue limit first, while also keeping a minimum of 2 queue items